File: ProyectoV2022/Probando/Repositories/MedicoRepository.cs
using ServicioMedico.Data;
using ServicioMedico.Entities;

namespace ServicioMedico.Repositories
{
    public class MedicoRepository : EFCoreRepository<Medico, DataContext>
    {
        private DataContext _dataContext;
        public MedicoRepository(DataContext context) : base(context)
        {
            _dataContext = context;
        }

        public async Task<Medico> createMedico(Medico medico)
        {
            List<Paciente> pacienteList = new List<Paciente>();
            foreach(Paciente p in medico.pacientes)
            {
                var paciente = await _dataContext.PacienteDB.FindAsync(p.Id);
                if (paciente != null)
                {
                    pacienteList.Add(paciente);
                }
            }
            
            medico.pacientes = pacienteList;
            medico.citas= new List<Cita>();
            return await this.add(medico);
        }
    }
}

File: ProyectoSpring/src/main/java/net/acc/customproyect/repositories/MedicoRepository.java
package net.acc.customproyect.repositories;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import net.acc.customproyect.entities.Cita;
import net.acc.customproyect.entities.Medico;

public interface MedicoRepository  extends JpaRepository<Medico, Long>{
	
}
